src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -50574,8 +50408,916 @@ class C {}
+@"
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesFalse(), type, "A<System.String>");
+        }
+
+        [Fact]
+        public void NonNullTypesContext_32()
+        {
+            var source =
+@"
+#pragma warning disable CS0169
+
+class A<T>
+{
+    void Tests()
+    {
+        string b;
+    }
+}
+";
+            TypeSyntax type = SyntaxFactory.ParseTypeName(
+@"
+#nullable disable
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesTrue(), type, "A<System.String>");
+        }
+
+        [Fact]
+        public void NonNullTypesContext_33()
+        {
+            var source =
+@"
+#pragma warning disable CS0169
+
+class A<T>
+{
+    void Tests()
+    {
+        string b;
+    }
+}
+";
+            TypeSyntax type = SyntaxFactory.ParseTypeName(
+@"
+#nullable enable
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesFalse(), type, "A<System.String!>");
+        }
+
+        [Fact]
+        public void NonNullTypesContext_34()
+        {
+            var source =
+@"
+#pragma warning disable CS0169
+
+class A<T>
+{
+    void Tests()
+    {
+        string 
+#nullable disable
+               b;
+#nullable enable
+    }
+}
+";
+            TypeSyntax type = SyntaxFactory.ParseTypeName(
+@"
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesTrue(), type, "A<System.String>");
+        }
+
+        [Fact]
+        public void NonNullTypesContext_35()
+        {
+            var source =
+@"
+#pragma warning disable CS0169
+
+class A<T>
+{
+    void Tests()
+    {
+        string 
+#nullable enable
+               b;
+#nullable disable
+    }
+}
+";
+            TypeSyntax type = SyntaxFactory.ParseTypeName(
+@"
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesFalse(), type, "A<System.String!>");
+        }
+
+        [Fact]
+        public void NonNullTypesContext_36()
+        {
+            var source =
+@"
+#pragma warning disable CS0169
+
+class A<T>
+{
+    void Tests()
+    {
+        string 
+#nullable disable
+               b;
+#nullable enable
+    }
+}
+";
+            TypeSyntax type = SyntaxFactory.ParseTypeName(
+@"
+#nullable enable
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesTrue(), type, "A<System.String!>");
+        }
+
+        [Fact]
+        public void NonNullTypesContext_37()
+        {
+            var source =
+@"
+#pragma warning disable CS0169
+
+class A<T>
+{
+    void Tests()
+    {
+        string 
+#nullable enable
+               b;
+#nullable disable
+    }
+}
+";
+            TypeSyntax type = SyntaxFactory.ParseTypeName(
+@"
+#nullable disable
+A<string>
+"
+            );
+
+            AssertNonNullTypesContext_30(source, WithNonNullTypesFalse(), type, "A<System.String>");
         }
```

Comment:
>} [](start = 8, length = 1)

Consider adding a test showing that the last token in a qualified type name is the one that determines the nullable context.

Reply:
> Consider adding a test showing that the last token in a qualified type name is the one that determines the nullable context. 

Current tests cover all the cases when the context is queried. Implementation doesn't depend on whether IdentifiernameSyntax,or GenericNamesyntax are quilified.

---
In reply to: [239971422](https://github.com/dotnet/roslyn/pull/31607#discussion_r239971422) [](ancestors = 239971422)